shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -1049,6 +1099,51 @@ - (FlutterPlatformViewController*)platformViewController {
 
 #pragma mark - Private methods
 
+- (void)addViewToEmbedderEngine:(FlutterViewController*)controller {
+  FlutterWindowMetricsEvent windowMetricsEvent{};
+  _populateMetricsEvent(controller, &windowMetricsEvent);
+  FlutterAddViewInfo info{
+      .struct_size = sizeof(FlutterAddViewInfo),
+      .view_id = controller.viewIdentifier,
+      .view_metrics = &windowMetricsEvent,
+      .add_view_callback =
+          [](const FlutterAddViewResult* result) {
+            // TODO(dkwingsmt): We should check if the result is successful, and
+            // block view operations until a successful response. For now we
+            // just assume all additions are successful. It works as of writing
+            // since the framework implemenation never returns failure, and
+            // messages are guaranteed in order.
+            FML_DCHECK(result->added);
+          },
+  };
+  FlutterEngineResult result = _embedderAPI.AddView(_engine, &info);
+  FML_DCHECK(result == kSuccess);
+}
+
+- (void)removeViewFromEmbedderEngine:(FlutterViewIdentifier)viewIdentifier {
+  FlutterRemoveViewInfo info{
+      .struct_size = sizeof(FlutterRemoveViewInfo),
+      .view_id = viewIdentifier,
+      .remove_view_callback =
+          [](const FlutterRemoveViewResult* result) {
+            // TODO(dkwingsmt): We should check if the result is successful, and
+            // hold deallocation until a successful response. For now we just
+            // assume all additions are successful. It works as of writing since
+            // the framework implemenation never returns failure, and messages
+            // are guaranteed in order.
+            FML_DCHECK(result->removed);
+          },
+  };
+  FlutterEngineResult result = _embedderAPI.RemoveView(_engine, &info);
+  FML_DCHECK(result == kSuccess);
+}
```

Comment:
Here's the Windows code that contains blocking logic you should be able  copy-paste: https://github.com/flutter/engine/blob/8b77056c74d6349a641b07a767e9f7eabb66414d/shell/platform/windows/flutter_windows_engine.cc#L573-L608